CONCEPT.md

```diff
@@ -6,25 +6,35 @@ Dieses Konzept stellt technische Lösungen zu den herausgearbeiteten Use-Cases d
 
 Nachfolgend wird allgemein für die nachfolgenden Titel einen technischen Lösungsansatz vorgestellt.
 
+### Pluginaufbau
+
+Das Plugin wird mehrere thematische Gebiete abdecken. Somit macht es Sinn, das Plugin in mehrere Bereiche zu trennen. Dies kann man einfach mit Unterordner im Root abbilden. Folgende Unterordner würden Sinn ergeben:
+
+- bergclub-touren
+- bergclub-adressen
+- bergclub-rueckmeldung
+- bergclub-export
+- bergclub-rechte
+
 ### Adressverwaltung
 
 Die Adressverwaltung wird im Plugin abgebildet. Dabei hat das Plugin folgende Aufgaben:
 
-1) Datenbankänderungen ([Link](https://codex.wordpress.org/Creating_Tables_with_Plugins)) für zusätzliche Felder, welche nicht als [User Metadata](https://developer.wordpress.org/plugins/users/working-with-user-metadata/) abgebildet werden können. 
-2) Exportfunktionen als [WP-Admin Menüpunkt.](https://developer.wordpress.org/plugins/administration-menus/)
-3) Funktion um eigene Adresse zu bearbeiten [Hier.](https://developer.wordpress.org/plugins/users/roles-and-capabilities/)
+1. Datenbankänderungen ([Link](https://codex.wordpress.org/Creating_Tables_with_Plugins)) für zusätzliche Felder, welche nicht als [User Metadata](https://developer.wordpress.org/plugins/users/working-with-user-metadata/) abgebildet werden können. 
+2. Exportfunktionen als [WP-Admin Menüpunkt.](https://developer.wordpress.org/plugins/administration-menus/)
+3. Funktion um eigene Adresse zu bearbeiten [Hier.](https://developer.wordpress.org/plugins/users/roles-and-capabilities/)
 
 ### Tourdaten
 
 Die Tourdaten werden mit Wordpress Posts inkl. Custom Fields abgearbeitet. Dabei hat das Theme folgende Aufgaben:
 
-1) Definieren der Custom Fields (siehe [Hier](https://developer.wordpress.org/plugins/metadata/))
-2) Korrektes Rendering der Metadaten.
+1. Definieren der Custom Fields (siehe [Hier](https://developer.wordpress.org/plugins/metadata/))
+2. Korrektes Rendering der Metadaten.
 
 Das Plugin hat folgende Aufgabe:
 
-1) Die Touren sollten mit dem korrekten Workflow bearbeitet werden. <!-- TODO -->
-2) Das Plugin definiert die Rollen und Rechte der Benutzer [Hier.](https://developer.wordpress.org/plugins/users/roles-and-capabilities/)
+1. Die Touren sollten mit dem korrekten Workflow bearbeitet werden ([Loop anpassen](https://codex.wordpress.org/The_Loop) / [Custom Post](https://developer.wordpress.org/plugins/post-types/)).
```

Comment:
Bin nicht 100% sicher aber glaube "The Loop" ist im Wordpress Theme der Loop fürs rendering, in welchem man sich befindet wenn durch Posts mit bestimmten properties (z.B. Kategorie) iteriert wird. Über dependency injection kann man dann auf die Post daten zugreifen.
Das Plugin welches den Workflow steuert, muss einen Funktion registrieren, die beim Speichern eines Posts aufgerufen wird:
https://codex.wordpress.org/Plugin_API/Action_Reference/save_post
Falls der Post in der Kategorie "Touren" oder "Touren Jugend" ist, muss es die zusätzlichen Metadaten sowie die Berechtigungen/Veröffentlichungsstatus validieren und gegebenenfalls das speichern unterbinden und in einer Funktion die dem admin_notices hook zugeordnet ist eine Fehlermeldung ausgeben.
https://codex.wordpress.org/Plugin_API/Action_Reference/admin_notices